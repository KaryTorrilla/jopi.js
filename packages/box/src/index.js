import styled from 'styled-components'
import {
  space,
  color,
  layout,
  flexbox,
  border,
  typography,
  position,
  variant,
} from 'styled-system'
import css from '@styled-system/css'

const sx = props => css(props.sx)(props.theme)
const base = props => css(props.__css)(props.theme)
const theme = ({ tx = 'variants' }) =>
  variant({
    scale: tx,
  })

export const Box = styled.div(
  {
    boxSizing: 'border-box',
    margin: 0,
    minWidth: 0,
  },
  base,
  theme,
  sx,
  space,
  color,
  layout,
  flexbox,
  typography,
  border,
  position
)

export const Flex = styled(Box)({
  display: 'flex',
})
