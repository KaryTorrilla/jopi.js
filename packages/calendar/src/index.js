import React from 'react'
import FullCalendar from '@fullcalendar/react' // must go before plugins
import dayGridPlugin from '@fullcalendar/daygrid' // a plugin!
import '@oneloop/theme/styles/globals.css'
import '../styles/styles.css'

export const Calendar = () => {
  /*   const calendarOptions = {
      events: [],
      plugins: [timeGridPlugin],
      initialView: 'timeGridDay',
      nowIndicator: true,
      headerToolbar: {
        start: 'prev',
        center: 'title',
        end: 'next',
      },
      editable: true,
      eventResizableFromStart: true,
      dragScroll: true,
      dayMaxEvents: 2,
      navLinks: true,
    } */

  return (
    <div className='calendarContainer'>
      <FullCalendar
        plugins={[dayGridPlugin]}
        initialView="dayGridMonth"
      />
    </div>
  )
}
