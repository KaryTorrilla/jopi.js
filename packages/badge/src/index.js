import React from 'react'
import { Box } from '@oneloop/box'
import '@oneloop/fonts'

export const Badge = ({ isMedium, hasIcon, isNotButton, ...props }) => (
  <Box
    as={ isNotButton ? "" : "button" }
    tx="buttons"
    variant="primary"
    {...props}
    __css={{
      lineHeight: '14px',
      height: isMedium ? '22px' : '16px',
      display: 'inline-block',
      fontSize: '10px',
      fontFamily: 'primary',
      textAlign: 'center',
      textTransform: 'uppercase',
      fontWeight: 'bold',
      letterSpacing: '0.05em',
      px: 2,
      py: '2px',
      color: 'white',
      bg: 'primary',
      border: 0,
      borderRadius: isMedium ? '14px' : '8px',
      flexDirection: 'row',
      justifyContent: 'center',
      alignItems: 'center',
      padding: hasIcon ? '4px 8px' : '1px 5px',
      position: 'static',
      flex: 'none',
      order: 0,
      flexGrow: 0,
      margin: '0px 0px',
    }}
  />
)
