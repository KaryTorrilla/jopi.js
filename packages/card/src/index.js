import React from 'react'
import { Box } from '@oneloop/box'
import theme from '@oneloop/theme'

export const Card = ({ variant, active, ...props }) => {
  const variantValues = Object.values(theme.card)[Object.keys(theme.card).indexOf(variant)]
  const backgroundColor = active ? variantValues[':focus'].backgroundColor : variantValues.backgroundColorValue
  return (
    <Box sx={{ position: 'relative' }}>
      <Box
        tx='card'
        variant={variant}
        {...props}
        __css={{
          fontFamily: 'primary',
          display: 'flex',
          flexDirection: 'row',
          alignItems: 'center',
          backgroundColor: backgroundColor,
          position: 'static',
        }}
        />
    </Box>
  )
}
