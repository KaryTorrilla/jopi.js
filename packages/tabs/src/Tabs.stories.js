import React from 'react'
import { Box } from '@oneloop/box'
import { Icon } from '@oneloop/icons'
import { Table } from '@oneloop/table'
import { Tabs } from '.'
import { Text } from '@oneloop/text'

export default {
  component: Tabs,
  title: 'Tabs',
  argTypes: {
    variant: {
      name: 'variant',
      description: 'Variantes disponibles del tab: [ \'normal\', \'default\' ]',
      type: 'text',
      control: { type: 'none' },
    },
    firstTabSelected: {
      name: 'firstTabSelected',
      description: 'Booleano que indica si se quiere mostrar el primer Tab seleccionado.',
      type: 'boolean',
      control: { type: 'none' },
    },
    variantBody: {
      name: 'variantBody',
      description: 'Variantes disponibles del texto: [ \'body\', \'body600\', \'bodyBold\' ]',
      type: 'text',
      control: { type: 'none' },
    },
    variantFont: {
      name: 'variantFont',
      description: 'Variantes disponibles de la fuente: [ \'fontSize09\', \'fontSize10\', \'fontSize11\', \'fontSize12\', \'fontSize13\', \'fontSize14\', \'fontSize15\', \'fontSize16\' ]',
      type: 'text',
      control: { type: 'none' },
    },
  },
}

export const normal = () => (
  <Tabs bg="primary" color="neutral.0" firstTabSelected={false}>
    <Tabs.Tab id="texto">
      <Box as="a" href="#">
        Texto
      </Box>
    </Tabs.Tab>
    <Tabs.Tab id="tabla">
      <Box as="a" href="#">
        Tabla
      </Box>
    </Tabs.Tab>
    <Tabs.Tab id="chau">
      <Box as="a" href="#">
        Chau
      </Box>
    </Tabs.Tab>
    <Tabs.Tab id="goodbye">
      <Box as="a" href="#">
        Goodbye
      </Box>
    </Tabs.Tab>
    <Tabs.Content id="texto">Hola!</Tabs.Content>
    <Tabs.Content id="tabla">
      <Table>
        <Table.Header>
          <Table.HeaderItem>Agency</Table.HeaderItem>
          <Table.HeaderItem>Country</Table.HeaderItem>
          <Table.HeaderItem>Paid</Table.HeaderItem>
        </Table.Header>
        <Table.Rows>
          <Table.Row>
            <Table.RowItem>InmoMex</Table.RowItem>
            <Table.RowItem>Mexico</Table.RowItem>
            <Table.RowItem>Yes</Table.RowItem>
          </Table.Row>
          <Table.Row>
            <Table.RowItem>RGM</Table.RowItem>
            <Table.RowItem>Argentina</Table.RowItem>
            <Table.RowItem>Yes</Table.RowItem>
          </Table.Row>
          <Table.Row>
            <Table.RowItem>Urbania</Table.RowItem>
            <Table.RowItem>Peru</Table.RowItem>
            <Table.RowItem>Yes</Table.RowItem>
          </Table.Row>
        </Table.Rows>
      </Table>
    </Tabs.Content>
  </Tabs>
)

export const tabDefault = () => (
  <Box __css={{ backgroundColor: 'lightgray', width: '250px', height: '80px', padding: '5px' }}>
    <Tabs className='grid gap-xs'>
      <Tabs.Tab variant='default' id="item0">
          <Icon icon='icon-buscar' fontSize='12px'/>
          <Text variantBody='body600' variantFont='fontSize12'>12</Text>
      </Tabs.Tab >
      <Tabs.Tab variant='default' id="item1">
          <Icon icon='icon-propiedades' fontSize='12px'/>
          <Text variantBody='body600' variantFont='fontSize12'>12</Text>
      </Tabs.Tab>
      <Tabs.Tab variant='default' id="item2">
          <Icon icon='icon-contactos' fontSize='12px'/>
          <Text variantBody='body600' variantFont='fontSize12'>12</Text>
      </Tabs.Tab>
      <Tabs.Tab variant='default' id="item3">
          <Icon icon='icon-emprendimientos' fontSize='12px'/>
          <Text variantBody='body600' variantFont='fontSize12'>12</Text>
      </Tabs.Tab>
      <Tabs.Content id="item0">Hola!</Tabs.Content>
      <Tabs.Content id="item1">Segundo item</Tabs.Content>
      <Tabs.Content id="item2">Tercer item</Tabs.Content>
      <Tabs.Content id="item3">Ultimo item</Tabs.Content>
    </Tabs>
  </Box>
)
