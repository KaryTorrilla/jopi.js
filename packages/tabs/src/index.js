import React, { useState, createContext, useContext } from 'react'
import { Box } from '@oneloop/box'

const TabsContext = createContext()

export const Tabs = ({ children, ...props }) => {
  const [active, setActive] = useState()
  const value = React.useMemo(() => ({ active, setActive }), [active])
  const tabChildren = React.Children.toArray(children).filter(
    child => child.type.name === 'Tab'
  )
  const contentChildren = React.Children.toArray(children).filter(
    child => child.type.name === 'Content'
  )

  return (
    <TabsContext.Provider value={value}>
      <Box {...props} __css={{ display: 'inline-flex' }}>
        {tabChildren}
      </Box>
      {contentChildren.length > 0 && (
        <Box __css={{ padding: '10px' }}>{contentChildren}</Box>
      )}
    </TabsContext.Provider>
  )
}

const useTabsContext = () => {
  const context = useContext(TabsContext)
  if (!context) {
    throw new Error(
      'Tabs compound components cannot be rendered outside the Tabs component'
    )
  }
  return context
}

const Tab = ({ id, children, ...props }) => {
  const { active, setActive } = useTabsContext()
  return (
    <Box
      onClick={() => setActive(id)}
      {...props}
      __css={{
        padding: 'auto 38px',
        position: 'relative',
        cursor: 'pointer',
        fontFamily: 'heading',
        textTransform: 'uppercase',
        textAlign: 'center',
        fontWeight: '600',
        fontSize: 1,
        '*': {
          textDecoration: 'none',
          color: active === id ? 'white' : '#EBA49A',
        },
        a: {
          display: 'block',
          width: '148px',
          height: '52px',
          lineHeight: '52px',
        },
        'a > *:first-child': {
          marginRight: '10px',
        },
      }}
    >
      {children}
      {active === id && (
        <Box
          as="span"
          __css={{
            borderRadius: '4px 4px 0 0',
            display: 'block',
            backgroundColor: 'white',
            height: '4px',
            position: 'absolute',
            right: '10px',
            left: '10px',
            bottom: 0,
          }}
        />
      )}
    </Box>
  )
}

const Content = ({ id, children, ...props }) => {
  const { active } = useTabsContext()
  return active === id && <Box {...props}>{children}</Box>
}

Tabs.Tab = Tab
Tabs.Content = Content
