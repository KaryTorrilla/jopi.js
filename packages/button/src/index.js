import React from 'react'
import { Badge } from '@oneloop/badge'
import { Box } from '@oneloop/box'
import { Checkbox } from '@oneloop/checkbox'
import { Icon } from '@oneloop/icons'
import { Image } from '@oneloop/image'
import { useToggle } from '@oneloop/hooks'
import theme from '@oneloop/theme'
import '@oneloop/fonts'

export const Button = ({ variant, ...props }) => (
  <Box sx={{ position: 'relative' }}>
    <Box
      as='button'
      tx='buttons'
      variant={variant}
      {...props}
      __css={{
        appearance: 'none',
        display: 'flex',
        lineHeight: 'inherit',
        fontFamily: 'Nunito Sans',
        fontWeight: 'normal',
        textAlign: 'center',
        textDecoration: 'none',
        cursor: 'pointer',
        outline: 'none',
        color: 'white',
        bg: 'primary',
        border: 0,
        borderRadius: '12px',
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        whiteSpace: 'nowrap',
      }}
    />
  </Box>
)

export const ButtonIcon = ({ icon, variant, text, badgeValue = 0, badgeVariant = 'primary', maxWidth, ...props }) => {
  let fontSize
  if (Array.isArray(variant)) {
    const indexes = variant.map(v => Object.keys(theme.buttons).indexOf(v))
    indexes.map(index => {
      if (Object.values(theme.buttons)[index].fontSize !== undefined) {
        fontSize = Object.values(theme.buttons)[index].fontSize
      }
      return fontSize
    })
  } else {
    const index = Object.keys(theme.buttons).indexOf(variant)
    fontSize = Object.values(theme.buttons)[index].fontSize
  }

  return (
    <Box sx={ maxWidth ? { position: 'relative', width: '100%' } : { position: 'relative' }}>
      <Box
        as='button'
        tx='buttons'
        variant={variant}
        {...props}
        __css={{
          appearance: 'none',
          display: 'flex',
          lineHeight: 'inherit',
          fontFamily: 'Nunito Sans',
          fontWeight: 'bold',
          textAlign: 'center',
          textDecoration: 'none',
          cursor: 'pointer',
          outline: 'none',
          border: 0,
          flexDirection: 'row',
          justifyContent: 'center',
          alignItems: 'center',
          whiteSpace: 'nowrap',
          position: 'relative',
        }}
      >
        { icon && <Icon icon={icon} fontSize={fontSize}/> }
        { badgeValue !== 0 && <Badge variant={badgeVariant} isNotButton style={{ position: 'absolute', top: '2px', left: '16px' }}>{ badgeValue }</Badge> }
        { text && <span>{text}</span> }
      </Box>
    </Box>
  )
}

export const ButtonHoldPress = ({ variant, active = false, text, icon, badgeValue = 0, badgeVariant = 'primary', isCollapsible, hasCheckbox = false, disabled = false, maxWidth, ...props }) => {
  let colorValue
  let colorValueIcon
  let fontSize
  let fontWeight
  if (Array.isArray(variant)) {
    const indexes = variant.map(v => Object.keys(theme.buttons).indexOf(v))
    indexes.map(index => {
      if (Object.values(theme.buttons)[index][':focus'] !== undefined && active) {
        colorValue = Object.values(theme.buttons)[index][':focus'].color
        fontWeight = Object.values(theme.buttons)[index][':focus'].fontWeight
        if (Object.values(theme.buttons)[index].colorIcon !== undefined) {
          colorValueIcon = Object.values(theme.buttons)[index][':focus'].colorIcon
        }
      } else if (Object.values(theme.buttons)[index].color !== undefined) {
        colorValue = Object.values(theme.buttons)[index].color
      } else if (Object.values(theme.buttons)[index].colorIcon !== undefined) {
        colorValueIcon = Object.values(theme.buttons)[index].colorIcon
      } else if (Object.values(theme.buttons)[index].fontWeight !== undefined) {
        fontWeight = Object.values(theme.buttons)[index].fontWeight
      }
      if (Object.values(theme.buttons)[index].fontSize !== undefined) {
        fontSize = Object.values(theme.buttons)[index].fontSize
      }
      return colorValue
    })
  } else {
    const index = Object.keys(theme.buttons).indexOf(variant)
    colorValue = Object.values(theme.buttons)[index].color
    colorValueIcon = Object.values(theme.buttons)[index].colorIcon
    fontSize = Object.values(theme.buttons)[index].fontSize
    fontWeight = Object.values(theme.buttons)[index].fontWeight
    if (active) {
      colorValue = Object.values(theme.buttons)[index][':focus'].color
      colorValueIcon = Object.values(theme.buttons)[index][':focus'].colorIcon
      fontWeight = Object.values(theme.buttons)[index][':focus'].fontWeight
    }
  }

  return (
    <Box sx={ maxWidth ? { position: 'relative', width: '100%' } : { position: 'relative' }}>
      <Box
        as='button'
        tx='buttons'
        variant={variant}
        {...props}
        __css={{
          appearance: 'none',
          display: 'flex',
          lineHeight: 'inherit',
          fontFamily: 'Nunito Sans',
          fontWeight: {fontWeight},
          textAlign: 'center',
          textDecoration: 'none',
          cursor: 'pointer',
          outline: 'none',
          border: 0,
          flexDirection: 'row',
          //  justifyContent: 'center',
          width: maxWidth &&'100%',
          alignItems: 'center',
          whiteSpace: 'nowrap',
        }}
      >
        { hasCheckbox && active && <Checkbox defaultChecked disabled={disabled}/>}
        { hasCheckbox && !active && <Checkbox defaultChecked={false} disabled={disabled}/>}
        { icon && <Icon icon={icon} fontSize={fontSize} style={{ color: ( colorValueIcon || colorValue ) }}/> }
        { text && <span style={colorValue && { color: colorValue, fontWeight: fontWeight }}>{text}</span>}
        { badgeValue === 0 && isCollapsible && <span className='icon-dropdown' style={{ position: 'absolute', right: '10px', color: '#798B97', fontSize: '14px', transform: active ? 'rotate(-180deg)' : 'rotate(0deg)', paddingTop: '4px' }}/> }
        { badgeValue !== 0 && !text && <Badge variant={badgeVariant} isNotButton style={{ position: 'absolute', top: '2px', left: '16px' }}>{badgeValue}</Badge> }
        { badgeValue !== 0 && text && <Badge variant={badgeVariant} isNotButton style={{ position: 'absolute', right: '16px' }}>{badgeValue}</Badge> }
      </Box>
    </Box>
  )
}

export const ButtonUser = ({ variant, srcImage, text, ...props }) => (
  <Box sx={{ position: 'relative' }}>
    <Box
      as='button'
      tx='buttons'
      variant={variant}
      {...props}
      __css={{
        appearance: 'none',
        display: 'flex',
        lineHeight: 'inherit',
        fontFamily: 'Nunito Sans',
        fontWeight: 'bold',
        fontSize: '14px',
        textAlign: 'center',
        textDecoration: 'none',
        cursor: 'pointer',
        outline: 'none',
        color: 'white',
        bg: 'primary',
        border: 0,
        borderRadius: '12px',
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
        height: '48px',
        whiteSpace: 'nowrap',
        padding: '13px 26px 13px 26px',
        gap: '8px',
      }}
    >
      {srcImage && <Image src={srcImage} variant="avatar"/> }
      {!srcImage && <span className='icon-contacto' style={{ fontSize: '24px', height: '24px' }}/>}
      {text}
    </Box>
  </Box>
)
