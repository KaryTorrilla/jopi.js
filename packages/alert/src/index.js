import React from 'react'
import { Flex } from '@oneloop/box'
import { Check, Exclamation, Times } from '@oneloop/icons'
import { Text } from '@oneloop/text'
import { Button } from '@oneloop/button'

export const Alert = ({ type = 'success', text, ...props }) => (
  <Flex
    variant="alert"
    {...props}
    __css={{
      width: '100%',
      bg: 'white',
      borderRadius: 'circle',
      boxShadow: 'active',
      justifyContent: 'space-between',
      p: 1,
    }}
  >
    {type === 'success' && (
      <Check
        bg="success"
        color="white"
        borderRadius="circle"
        size={40}
        p={3}
        mr={5}
      />
    )}
    {type === 'warning' && (
      <Exclamation
        bg="warning"
        color="white"
        borderRadius="circle"
        size={40}
        p={3}
        mr={5}
      />
    )}
    {type === 'danger' && (
      <Times
        bg="danger"
        color="white"
        borderRadius="circle"
        size={40}
        p={3}
        mr={5}
      />
    )}
    <Text variant="body.0" display="flex" alignItems="center" flexGrow={1}>
      {text}
    </Text>
    <Button
      size={32}
      bg="#F6F2F2"
      p={3}
      display="inline-flex"
      alignSelf="center"
      mr={2}
    >
      <Times color="#716F6F" m="auto" />
    </Button>
  </Flex>
)
