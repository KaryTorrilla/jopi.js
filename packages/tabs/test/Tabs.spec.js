import React from 'react'
import renderer from 'react-test-renderer'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import 'jest-styled-components'

import { Tabs } from '../src'

Enzyme.configure({ adapter: new Adapter() })

describe('Tabs', () => {
  test('renders correctly with only headers', () => {
    const tree = renderer
      .create(
        <Tabs>
          <Tabs.Tab id="one">One</Tabs.Tab>
          <Tabs.Tab id="two">Two</Tabs.Tab>
          <Tabs.Tab id="three">Three</Tabs.Tab>
          <Tabs.Tab id="four">Four</Tabs.Tab>
        </Tabs>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('renders correctly with headers and content', () => {
    const tree = renderer
      .create(
        <Tabs>
          <Tabs.Tab id="one">One</Tabs.Tab>
          <Tabs.Tab id="two">Two</Tabs.Tab>
          <Tabs.Tab id="three">Three</Tabs.Tab>
          <Tabs.Tab id="four">Four</Tabs.Tab>
          <Tabs.Content id="one">One Content One</Tabs.Content>
          <Tabs.Content id="three">Three Content Three</Tabs.Content>
        </Tabs>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('renders correctly with variant default', () => {
    const component = shallow(
      <Tabs>
        <Tabs.Tab variant='default' id="one" onClick={console.log('works')}>One</Tabs.Tab>
        <Tabs.Tab variant='default' id="two">Two</Tabs.Tab>
        <Tabs.Tab variant='default' id="three">Three</Tabs.Tab>
        <Tabs.Tab variant='default' id="four">Four</Tabs.Tab>
        <Tabs.Content id="one">One Content One</Tabs.Content>
        <Tabs.Content id="three">Three Content Three</Tabs.Content>
      </Tabs>
    )
    expect(component.find(Tabs.Tab).at(0).exists).toBeTruthy()

    component.find(Tabs.Tab).at(0).simulate('mouseover')
    component.find(Tabs.Tab).at(0).simulate('mouseout')
    component.find(Tabs.Tab).at(0).simulate('focus')
    component.find(Tabs.Tab).at(0).simulate('blur')
    component.find(Tabs.Tab).at(0).simulate('click')

    expect(component).toMatchSnapshot()
  })

  test('renders throw error', () => {
    try {
      const tree = () => {
        renderer
          .create(
            <Tabs.Tab>
            </Tabs.Tab>
          )
          .toJSON()
      }
      expect(tree).toThrow(Error)
    } catch (e) {
      expect(e).toThrow(Error)
    }
  })
})
