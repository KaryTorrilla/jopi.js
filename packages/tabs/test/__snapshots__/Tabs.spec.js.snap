// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs renders correctly with headers and content 1`] = `
Array [
  .c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: auto 38px;
  position: relative;
  cursor: pointer;
  font-family: heading;
  text-transform: uppercase;
  text-align: center;
  font-weight: 600;
  font-size: 14px;
}

.c1 * {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #EBA49A;
}

.c1 a {
  display: block;
  width: 148px;
  height: 52px;
  line-height: 52px;
}

.c1 a > *:first-child {
  margin-right: 10px;
}

<div
    className="c0"
  >
    <div
      className="c1"
      onClick={[Function]}
    >
      One
    </div>
    <div
      className="c1"
      onClick={[Function]}
    >
      Two
    </div>
    <div
      className="c1"
      onClick={[Function]}
    >
      Three
    </div>
    <div
      className="c1"
      onClick={[Function]}
    >
      Four
    </div>
  </div>,
  .c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 10px;
}

<div
    className="c0"
  />,
]
`;

exports[`Tabs renders correctly with only headers 1`] = `
Array [
  .c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: auto 38px;
  position: relative;
  cursor: pointer;
  font-family: heading;
  text-transform: uppercase;
  text-align: center;
  font-weight: 600;
  font-size: 14px;
}

.c1 * {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #EBA49A;
}

.c1 a {
  display: block;
  width: 148px;
  height: 52px;
  line-height: 52px;
}

.c1 a > *:first-child {
  margin-right: 10px;
}

<div
    className="c0"
  >
    <div
      className="c1"
      onClick={[Function]}
    >
      One
    </div>
    <div
      className="c1"
      onClick={[Function]}
    >
      Two
    </div>
    <div
      className="c1"
      onClick={[Function]}
    >
      Three
    </div>
    <div
      className="c1"
      onClick={[Function]}
    >
      Four
    </div>
  </div>,
  .c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 10px;
}

<div
    className="c0"
  />,
]
`;
